src/threading.ts

```diff
@@ -117,12 +118,33 @@ class Thread {
     public writeTraceback(): void {
         this.traceback = "Traceback (most recent call last):\n" + this.traceback;
         process.stdout.write(this.traceback);
+        this.exit();
     }
 
-    // Output the traceback when thread cannot handle an exception, and exit the thread
-    public throwException(): void {
-        this.writeTraceback();
-        this.exit();
+    public throwException(f: Py_FrameObject): void {
```

Comment:
But `throwException` will also be called by native code. There's no frame stack in that case. And native functions don't have a frame stack, either.

`throwException` should abstract away the specific details involved with exception handling. When I call `thread.throwException(e)` from code I'm writing, I shouldn't have to think about setting up a frame stack first.

If the exception needs to be in the stack for `COMPARE_OP`, you should be pushing it from `tryCatchException`.